src/modules/types/science/biology/growth_plate/view.js

```diff
@@ -0,0 +1,695 @@
+        this.samplesList = null;
+        API.killHighlight(this.module.getId());
+        this.dom.empty();
+      },
+      cellsList: function () {
+        this.plate = null;
+        this.cellsList = null;
+        API.killHighlight(this.module.getId());
+        this.dom.empty();
+      },
+      samplesList: function () {
+        this.plate = null;
+        this.cellsList = null;
+        this.samplesList = null;
+        API.killHighlight(this.module.getId());
+        this.dom.empty();
+      },
+    },
+
+    inDom: function () {
+      const that = this;
+      this.dom.on('mouseenter mouseleave click', 'td', function (e) {
+        const cellsList = that.cellsList,
+          plate = $(this).parents(':eq(3)').index(),
+          trIndex = $(this).parent().index(),
+          tdIndex = $(this).index(),
+          cols = that.cols,
+          rows = that.rows,
+          direction = that.module.getConfiguration('direction', 'vertical') || 'vertical',
+          elementId = direction === 'vertical' ?
+            (plate * cols * rows) + tdIndex * rows + trIndex :
+            (plate * cols * rows) + trIndex * cols + tdIndex;
+        if (!cellsList[elementId]) return;
+        let highlight = cellsList[elementId]._highlight;
+        if (e.type === 'mouseenter') {
+          that.module.controller.createDataFromEvent(
+            'onTrackMouse',
+            'trackData',
+            cellsList[elementId],
+          );
+          that.module.controller.sendActionFromEvent(
+            'onTrackMouse',
+            'trackData',
+            cellsList[elementId],
+          );
+          API.highlight([highlight], 1);
+        } else if (e.type === 'mouseleave') {
+          API.highlight([highlight], 0);
+        } else if (e.type === 'click') {
+          that.module.controller.sendActionFromEvent(
+            'onTrackClick',
+            'trackData',
+            cellsList[elementId],
+          );
+        }
+      });
+      this.resolveReady();
+    },
+
+    update: {
+      plate: function (moduleValue) {
+        const cfg = this.module.getConfiguration,
+          cfgc = this.module.getConfigurationCheckbox,
+          cols = cfg('colnumber', 4) || 4,
+          rows = cfg('rownumber', 4) || 4,
+          style = cfg('shape', 'style2') || 'style2',
+          direction = cfg('direction', 'vertical') || 'vertical',
+          random = cfg('random', 'sequential') || 'sequential',
+          colorJpath = cfg('colorjpath', false),
+          val = moduleValue.get(),
+          colorBySample = cfgc('colorBySample', 'yes'),
+          colorByJpathValue = cfgc('colorByJpathValue', 'yes'),
+          colorByJpath = cfgc('colorByJpath', 'yes'),
+          replicates = val.replicates,
+          mode = random === 'random' ? true : false,
+          control = val.control ? Object.entries(val.control) : false;
+        this.plate = val;
+        let shape;
+        switch (style) {
+          case 'style1': {
```

Comment:
style1, style2, style3 please try to get better names to describe the style like aligned, shifted, shifted vertically